### PR TITLE
controllers setup

### PR DESCRIPTION
**Pull request by nxrara**
- controllers setup and seeding

### CODE REVIEW

File: app/controllers/trips_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class TripsController < ApplicationController
+  # GET /trips
+  def index
+    @trips = Trip.where(users: { id: current_user.id })
+  end
+
+  # POST /trails/:trail_id/trips
+  def create
+    @trail = Trail.find(params[:trail_id])
+    trip = Trip.new(user: current_user, trail: trail)
```
Comment (barbayjuliette) [inline app/controllers/trips_controller.rb:10]: should be @trail

File: app/models/trip.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
 class Trip < ApplicationRecord
+  belongs_to :user
+  belongs_to :trail
+
+  validates :date, presence: true
```
Comment (barbayjuliette) [inline app/models/trip.rb:5]: validation on format. Has to be a date format

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -1,25 +1,46 @@
 <h2>Sign up</h2>
 
-<%= simple_form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+<div class="main-title">
+  <h2 class="text-center">Make an account</h2>
+  <h6 class="login-text">Order or sell, and view your order history by making an account.</h6>
```
Comment (barbayjuliette) [inline app/views/devise/registrations/new.html.erb:5]: Change text

File: config/routes.rb [hunk 4]
```diff
@@ -5,4 +5,12 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  resources :trails, only: %i[index show] do
+    resources :trips, only: :create
```
Comment (barbayjuliette) [inline config/routes.rb:10]: we will also need destroy to remove a trip. Also show and index and update

**Comment by barbayjuliette** [comment]
For the seed with faker, for the ones that have validation unique, you have to specify unique in faker, otherwise can bug

**Comment by barbayjuliette** [comment]
Use .sample instead of .shuffle.first

**Comment by barbayjuliette** [comment]
Root path doesn't work: The action 'index' could not be found for PagesController

**Review by lxingern — APPROVED** [review]
Ok, all comments seem to have been resolved

**Pull request merged**